### PR TITLE
Homework#3

### PR DESCRIPTION
**Pull request by Petrus9107**

### CODE REVIEW

File: module3/homework/passwordCheck/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,69 @@
...
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+
+std::string getErrorMessage(const ErrorCode& code) {
+    switch (code) {
+    case ErrorCode::Ok:
+        return "OK";
+        break;
```
Comment (DamiXioR) [inline module3/homework/passwordCheck/validation.cpp:13]: in my opinion break is redundant when you use return before

File: module3/homework/passwordCheck/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,69 @@
...
+}
+
+bool doesPasswordsMatch(const std::string& pass1, const std::string& pass2) {
+    return pass1 == pass2;
+}
+
+ErrorCode checkPasswordRules(const std::string& pass) {
+    constexpr int PASSWORD_LENGTH = 9;
+
+    if (pass.length() < PASSWORD_LENGTH)
```
Comment (DamiXioR) [inline module3/homework/passwordCheck/validation.cpp:47]: looks better when you use curly braces with every if statement

File: module3/homework/passwordCheck/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,69 @@
...
+        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+    else {
+        return ErrorCode::Ok;
+    }
+}
+
+ErrorCode checkPassword(const std::string& pass1, const std::string& pass2) {
+    if (!doesPasswordsMatch(pass1, pass2))
+        return ErrorCode::PasswordsDoesNotMatch;
+    else {
```
Comment (DamiXioR) [inline module3/homework/passwordCheck/validation.cpp:66]: redundant else

File: module3/homework/passwordCheck/validation.hpp [hunk 4]
```diff
@@ -1,2 +1,18 @@
...
+#include <iostream>
+#include <string>
+
+enum class ErrorCode {
+    Ok,
+    PasswordNeedsAtLeastNineCharacters,
+    PasswordNeedsAtLeastOneNumber,
+    PasswordNeedsAtLeastOneSpecialCharacter,
+    PasswordNeedsAtLeastOneUppercaseLetter,
+    PasswordsDoesNotMatch,
```
Comment (DamiXioR) [inline module3/homework/passwordCheck/validation.hpp:11]: you don't need last comma

**Comment by ziobron** [comment]
+7 XP Podstawy#3 - passwordCheck (1995)
+4 XP Podstawy#3 – passwordCheck bonus requirements (1996)

{ "requirement_ids": [1995, 1996], "points": [7, 4] }

Should we close this PR or you want to deliver vectorOfSharedPointers as well? :)

File: module3/homework/passwordCheck/validation.cpp (hunk 3 above)
Comment (nauka-programowania-MA) [inline module3/homework/passwordCheck/validation.cpp:66]: else is redundant here :)

File: module3/homework/passwordCheck/validation.hpp (hunk 4 above)
Comment (nauka-programowania-MA) [inline module3/homework/passwordCheck/validation.hpp:11]: remove last `,`

File: module3/homework/passwordCheck/validation.cpp [hunk 5]
```diff
@@ -1,2 +1,69 @@
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+
+#include <stdlib.h>
+
+#include <algorithm>
+#include <iostream>
+#include <string>
```
Comment (WBigaj) [inline module3/homework/passwordCheck/validation.cpp:7]: You have included <string> and <iostream> in the header.No need to include them in cpp file.

File: module3/homework/passwordCheck/validation.cpp (hunk 3 above)
Comment (WBigaj) [reply module3/homework/passwordCheck/validation.cpp:66]: When you return value the function is over,so there is no need to play with else if structure. The last else is defintely redundant.

File: module3/homework/passwordCheck/validation.hpp [hunk 6]
```diff
@@ -1,2 +1,18 @@
...
+
+};
+
+std::string getErrorMessage(const ErrorCode& code);
+bool doesPasswordsMatch(const std::string& pass1, const std::string& pass2);
+ErrorCode checkPasswordRules(const std::string& pass);
+ErrorCode checkPassword(const std::string& pass1, const std::string& pass2);
```
Comment (ziobron) [inline module3/homework/passwordCheck/validation.hpp:18]: sorry, tactical comment. Testing purposes only ;)

**Comment by ziobron** [comment]
Let me close this PR. If you wish to deliver something else please re-open it.

**Pull request closed without merging**